src/GitVersionCore.Tests/BuildServers/VsoAgentTests.cs

```diff
@@ -1,16 +1,35 @@
-﻿using GitVersion;
+﻿using System;
+using GitVersion;
+using GitVersionCore.Tests;
 using NUnit.Framework;
 using Shouldly;
 
 [TestFixture]
 public class VsoAgentTests
 {
+
+    string key = "BUILD_BUILDNUMBER";
+
+    private void SetEnvironmentVariableForTest()
+    {
+        Environment.SetEnvironmentVariable(key, "Some Build_Value $(GitVersion_FullSemVer)", EnvironmentVariableTarget.Process);
```

Comment:
Maybe it would be better to test with some more real life like example for "Some Build_Value" (e.g use the default value "$(date:yyyyMMdd)$(rev:.r)")